src/System.Transactions.Local/src/System/Transactions/TransactionState.cs

```diff
@@ -21,719 +21,144 @@ internal abstract class TransactionState
         // state transition is to active.  All other state transitions must happen within the state machines 
         // themselves.
         // Double-checked locking pattern requires volatile for read/write synchronization
```

Comment:
>volatile [](start = 51, length = 8)

I am not clear on why the volatile keyword can be removed here. How does LazyInitializer.EnsureInitialized avoid requiring it to be volatile?

If it is, indeed, safe to remove volatile, then this comment should be removed.